main/src/ca/uwaterloo/flix/api/Bootstrap.scala

```diff
@@ -699,4 +701,92 @@ class Bootstrap(val projectPath: Path, apiKey: Option[String]) {
 
     Validation.success(())
   }
+
+  /**
+    * Show dependencies which have newer versions available.
+    */
+  def outdated(flix: Flix)(implicit out: PrintStream): Validation[Unit, BootstrapError] = {
```

Comment:
There seems to be a lot going on here. A few suggestions:
- We can separate the "getting the data" part and the "formatting the data" part into their own functions.
- We might be able to clean up some of the match-on-Result stuff with a `for`.
- Comments describing what the major/minor/patch stuff does would help.
- Maybe major/minor/patch stuff could be part of the SemanticVersion namespace?